src/openforms/js/components/admin/forms/MappingArrayInput.js

```diff
@@ -24,20 +25,24 @@ const MappingArrayInput = ({
     const newMapping = produce(mapping, draft => {
       draft.push(['', newValue]);
     });
-    onChange(newMapping);
+    onChange(name, newMapping);
   };
 
   const onDelete = index => {
-    return produce(mapping, draft => draft.filter((_, itemIndex) => itemIndex !== index));
+    const newMapping = produce(mapping, draft =>
+      draft.filter((_, itemIndex) => itemIndex !== index)
+    );
+    onChange(name, newMapping);
```

Comment:
@sergei-maertens produce didn't seem to actually lead to anything being deleted, I'm still unsure of how this should all work together, because the state is initialized by Formik in the `ServiceConfigurationFormPicker`, but at the same time it seems as though React components should also keep track of their own state